File: Count_All_Valid_Pickup_and_Delivery_Options.java
class Solution {
    public int countOrders(int n) {
        final int MOD = 1000000007;
        int totalWays = 1;
        for (int orderNumber = 2; orderNumber <= n; orderNumber++) {
            totalWays = (int)((totalWays * (2L * orderNumber - 1) * orderNumber) % MOD);
        }
        return totalWays;        
    }
}

File: Count_All_Valid_Pickup_and_Delivery_Options.py
class Solution:
    def countOrders(self, n: int) -> int:
        MOD = 10**9 + 7
        total_ways = 1

        for order_number in range(2, n + 1):
            total_ways = (total_ways * (2 * order_number - 1) * order_number) % MOD
        
        return total_ways
